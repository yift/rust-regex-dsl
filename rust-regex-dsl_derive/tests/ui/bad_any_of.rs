pub use rust_regex_dsl::regex_dsl;

fn main() {
    let regex = regex_dsl! {
        any_of,
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of(),
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of(""),
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            not_a_word
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            #not_a_class
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            from, 'a'
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            from: 'a' tof: 'b'
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            from: 'a' to, 'b'
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            from: 'a' to: 1
        },
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        any_of {
            from: 1 to: 'a'
        },
    };
    println!("{}", regex);
}