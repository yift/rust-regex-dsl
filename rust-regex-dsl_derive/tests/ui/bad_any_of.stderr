error: Unknown word: any_of
 --> tests/ui/bad_any_of.rs:5:9
  |
5 |         any_of,
  |         ^^^^^^

error: unexpected end of input, Empty class is not supported
 --> tests/ui/bad_any_of.rs:9:16
  |
9 |         any_of(),
  |                ^

error: Empty string is not supported
  --> tests/ui/bad_any_of.rs:13:16
   |
13 |         any_of(""),
   |                ^^

error: Unknown word: not_a_word
  --> tests/ui/bad_any_of.rs:18:13
   |
18 |             not_a_word
   |             ^^^^^^^^^^

error: Unknown regex class not_a_class
  --> tests/ui/bad_any_of.rs:24:14
   |
24 |             #not_a_class
   |              ^^^^^^^^^^^

error: expected `:`
  --> tests/ui/bad_any_of.rs:30:17
   |
30 |             from, 'a'
   |                 ^

error: Expecting to
  --> tests/ui/bad_any_of.rs:36:23
   |
36 |             from: 'a' tof: 'b'
   |                       ^^^

error: expected `:`
  --> tests/ui/bad_any_of.rs:42:25
   |
42 |             from: 'a' to, 'b'
   |                         ^

error: expected character literal
  --> tests/ui/bad_any_of.rs:48:27
   |
48 |             from: 'a' to: 1
   |                           ^

error: expected character literal
  --> tests/ui/bad_any_of.rs:54:19
   |
54 |             from: 1 to: 'a'
   |                   ^

error: Missing content
  --> tests/ui/bad_any_of.rs:62:17
   |
62 |             xor "aa"
   |                 ^^^^

error: expected one of: string literal, character literal, `#`, `~`, identifier
  --> tests/ui/bad_any_of.rs:69:17
   |
69 |             xor(5)
   |                 ^
